packages/deploy/src/client.ts

```diff
@@ -37,6 +42,30 @@ export async function getProject(
   }
 }
 
+export async function downloadSpec(
+  config: DeployConfig,
+  projectId: string
+): Promise<{ data: ProjectPayload | null }> {
+  let base = new URL(config.endpoint).origin; 
+  let url = new URL(base + `/download/yaml`);
+  url.searchParams.set('id', projectId);
+
+
+  try {
+      const res = await fetch(url);
+      const destination = path.resolve(".", config.specPath);
+      const fileStream = fs.createWriteStream(destination, {flags: 'wx'});
```

Comment:
I've not tested this but I'm pretty sure you can stream the reponse to file like this:
```
// import like this
import fs from 'node:fs/promises';

// back inside downloadSpec
const res = await fetch(url);
return fs.writeFile(path.resolve(config.specPath), res.body)
```

`writeFile` can take a stream as the input data, and `Response.body` returns a readable stream. So it should jsut work!